File: API/PageAndRDLX/FontResolver/C#/FontResolver/MainForm.cs
using GrapeCity.ActiveReports.Export.Pdf.Page;
using GrapeCity.ActiveReports.Rendering.IO;
using ActiveReports.Samples.FontResolver.Properties;
using System;
using System.IO;
using System.Windows.Forms;
using Settings = GrapeCity.ActiveReports.Export.Pdf.Page.Settings;
using GrapeCity.ActiveReports;

namespace ActiveReports.Samples.FontResolver
{
	public partial class MainForm : Form
	{
		PageReport _pageReport = new PageReport();

		/// <summary>
		/// Form Constructor.
		/// </summary>
		public MainForm()
		{
			InitializeComponent();
		}

		/// <summary>
		/// Preview the Layer report and set the target device for the layer to the report.
		/// </summary>
		/// <param name="sender"></param>
		/// <param name="e"></param>
		private void btnPreview_Click(object sender, EventArgs e)
		{
			_pageReport.Load(new FileInfo("../../../../../../Reports/Barcodes.rdlx"));
			_pageReport.FontResolver = BarcodeFontResolver.Instance;
			reportViewer.LoadDocument(_pageReport.Document);
			btnPdfExport.Enabled = true;
		}

		/// <summary>
		/// Export the report displayed in Viewer to PDF.
		/// </summary>
		/// <param name="sender"></param>
		/// <param name="e"></param>
		private void btnPdfExport_Click(object sender, EventArgs e)
		{
			if (_pageReport != null)
			{
				Settings settings = new Settings();
				settings.HideToolbar = true;
				settings.HideMenubar = true;
				settings.HideWindowUI = true;
				saveFileDialog.Filter = Resources.PDFFilter;

				PdfRenderingExtension _renderingExtension = new PdfRenderingExtension();
				if (saveFileDialog.ShowDialog() == DialogResult.OK)
				{
					if (File.Exists(saveFileDialog.FileName))
						File.Delete(saveFileDialog.FileName);

					FileStreamProvider _exportfile = new FileStreamProvider(new DirectoryInfo(Path.GetDirectoryName(saveFileDialog.FileName)), Path.GetFileNameWithoutExtension(saveFileDialog.FileName));
					_pageReport.Document.Render(_renderingExtension, _exportfile, settings);
				}
			}
		}
	}
}

File: API/PageAndRDLX/FontResolver/VB.NET/FontResolver/MainForm.vb
Imports GrapeCity.ActiveReports.Rendering.IO
Imports System.IO
Imports System.Text
Imports GrapeCity.ActiveReports.Export.Pdf.Page
Imports GrapeCity.ActiveReports

Public Class MainForm

    Private _pageReport As New PageReport()

    'Constructor.
    Sub New()
        Encoding.RegisterProvider(CodePagesEncodingProvider.Instance)
        InitializeComponent()
    End Sub

    Private Sub btnPreview_Click(sender As Object, e As EventArgs) Handles btnPreview.Click
        _pageReport.Load(New FileInfo("../../../../../../Reports/Barcodes.rdlx"))
        _pageReport.FontResolver = BarcodeFontResolver.Instance
        reportViewer.LoadDocument(_pageReport.Document)
        btnPdfExport.Enabled = True
    End Sub

    'Export the report displayed in Viewer to PDF.
    Private Sub btnPdfExport_Click(sender As Object, e As EventArgs) Handles btnPdfExport.Click
        If _pageReport IsNot Nothing Then
            Dim settings As New Settings()
            settings.HideToolbar = True
            settings.HideMenubar = True
            settings.HideWindowUI = True
            saveFileDialog.Filter = My.Resources.Resources.PDFFilter

            Dim renderingExtension As New PdfRenderingExtension()
            If saveFileDialog.ShowDialog() = DialogResult.OK Then
                If File.Exists(saveFileDialog.FileName) Then
                    File.Delete(saveFileDialog.FileName)
                End If

                Dim exportFile As New FileStreamProvider(New DirectoryInfo(Path.GetDirectoryName(saveFileDialog.FileName)), Path.GetFileNameWithoutExtension(saveFileDialog.FileName))
                _pageReport.Document.Render(renderingExtension, exportFile, settings)
            End If
        End If
    End Sub

End Class
